machine.go

```diff
@@ -424,15 +391,53 @@ func (m *Machine) setupLogging(ctx context.Context) error {
 	}
 
 	resp, err := m.client.PutLogger(ctx, &l)
-	if err == nil {
-		m.logger.Printf("Configured VMM logging to %s, metrics to %s: %s",
-			m.cfg.LogFifo, m.cfg.MetricsFifo, resp.Error())
+	if err != nil {
+		return err
 	}
-	return err
+
+	m.logger.Printf("Configured VMM logging to %s, metrics to %s: %s",
+		m.cfg.LogFifo, m.cfg.MetricsFifo, resp.Error())
+
+	if m.cfg.FifoLogWriter != nil {
+		if err := captureFifoToFile(m.logger, m.cfg.LogFifo, m.cfg.FifoLogWriter); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func captureFifoToFile(logger *log.Entry, fifoPath string, fifo io.Writer) error {
+	// create the fifo pipe which will be used
+	// to write its contents to a file.
+	fifoPipe, err := os.OpenFile(fifoPath, os.O_RDONLY, 0600)
+	if err != nil {
+		return fmt.Errorf("Failed to open fifo path at %q: %v", fifoPath, err)
+	}
+
+	if err := syscall.Unlink(fifoPath); err != nil {
+		logger.Warnf("Failed to unlink %s", fifoPath)
+	}
+
+	logger.Infof("Capturing %q to writer", fifoPath)
```

Comment:
Let's drop this to Debug severity. We'll do a more complete log cleanup later, but for now let's get new messages in the right place to begin with.